packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,287 @@
+            tabPanel: get(this.props, "managedClasses.pivot_tabPanel", ""),
+            tabPanel__hidden: get(
+                this.props,
+                "managedClasses.pivot_tabPanel__hidden",
+                ""
+            ),
+        };
+    }
+
+    /**
+     * Adds focus state to outer wrapper
+     * In order perfomantly apply focus to activeIndicator,
+     * a class must be added instead of using
+     * focus-visual via style
+     */
+    private handleOnFocus = (): void => {
+        const tabElement: HTMLElement = ReactDOM.findDOMNode(
+            this.tabsRef.current
+        ) as HTMLElement;
+
+        const mytabs: NodeListOf<Element> = tabElement.querySelectorAll(
+            canUseFocusVisible() ? "[role='tab']:focus-visible" : "[role='tab']:focus"
+        );
+
+        if (mytabs.length > 0) {
+            this.setState({ focused: true });
+        }
+    };
+
+    /**
+     * Removes focus state
+     */
+    private handleOnBlur = (): void => {
+        this.setState({ focused: false });
+    };
+
+    /**
+     * Generates class names
+     */
+    private generateActiveIndicatorClassNames(): string {
+        let classNames: string = get(
+            this.props,
+            "managedClasses.pivot_activeIndicator",
+            ""
+        );
+
+        if (this.state.focused) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.pivot_activeIndicator__focused",
+                ""
+            )}`;
+        }
+
+        return classNames;
+    }
+
+    private handleTabsUpdate = (activeTabId: string): void => {
+        this.updateTabPanelIndex();
+        if (this.props.activeId) {
+            this.props.onUpdate(activeTabId);
+        } else {
+            this.setState(
+                {
+                    activeId: activeTabId,
+                },
+                this.setActiveIndicatorOffset
+            );
+        }
+    };
+
+    private isSelected(element: HTMLElement): boolean {
+        return element.className.includes("active") === true;
+    }
+
+    private updateTabPanelIndex(): number {
+        if (canUseDOM() && this.tabsRef.current && this.props.items) {
+            const tabElement: HTMLElement = ReactDOM.findDOMNode(
+                this.tabsRef.current
+            ) as HTMLElement;
+
+            const mytabsArray: HTMLElement[] = Array.prototype.slice.call(
+                tabElement.querySelectorAll("[role='tab']")
+            );
+
+            return mytabsArray.findIndex(this.isSelected);
+        }
+    }
+
+    private generateTabPanelsClassNames(): string {
+        let className: string = get(this.props, "managedClasses.pivot_tabPanels", "");
+        if (this.props.items) {
+            if (this.tabPanelIndex === this.updateTabPanelIndex()) {
+                className = className;
+            } else if (this.tabPanelIndex < this.updateTabPanelIndex()) {

```

Comment:
what is the initial "if" doing? Do we need it or we can rewrite this in a way that removes that complexity?

Reply:
can be removed since I've now made items a required prop